### PR TITLE
Add Postgres compliant name aliasing for String Functions.

### PR DESCRIPTION
**Pull request by mayankshriv**
Certain String function names are not Postgres compliant, added aliasing to fix that.
- left
- right
- split_part
- string_to_array

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java [hunk 1]
```diff
@@ -243,7 +243,7 @@ public static String rtrim(String input) {
    * @param input
    * @return get substring starting from the first index and extending upto specified length.
    */
-  @ScalarFunction
+  @ScalarFunction(names = {"leftSubStr", "left"})
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java:246]: Is `leftSubStr` a valid function in any common DB? Should we add the underscore version?

**Review by Jackie-Jiang — APPROVED** [review]
Would be good to add a test for both V1 and V2

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java (hunk 1 above)
Comment (mayankshriv) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java:246]: I had checked, with v1, all variations are supported `left_substr`, `leftSubStr`, `left_sub_str`, which is why I didn't add them explicitly. For v2, we want to be Postgres compliant, so there is just `left`.

**Comment by mayankshriv** [comment]
> Would be good to add a test for both V1 and V2

Yes, agree. I don't see any existing tests for most of these functions. If we want to test aliasing, it can't be unit test for the functions, it has to go through sql compilation. This can be a followup.

**Pull request merged**